src/topology.rs

```diff
@@ -9,6 +9,7 @@ pub struct Topology {
     nodes: Vec<NodeId>,                  // The list of nodes in the topology
     edges: HashMap<NodeId, HashSet<NodeId>>, // All the connections between nodes.
     labels: HashMap<NodeId, String>,     // The labels of the nodes
```

Comment:
Sorry but for the labels, how do we need to handle them? 
I thought the labels were the NodeType.to_string(), but since you are adding another map for that, what do we need to set as values?

Reply:
I was thinking only controller side, when creating this structures, but you are right. 
The type should be what you are using as NodeType.to_string()
The label instead is mostly presentation side to give a Label to the nodes  which might be different from their type and id.